controllers/configurationpolicy_controller.go

```diff
@@ -1914,16 +1917,34 @@ func mergeArrays(newArr []interface{}, old []interface{}, ctype string) (result
 			}
 
 			var mergedObj interface{}
+			// Stores if val1 and val2 are maps with the same "name" key value. In the case of the containers array
+			// in a Deployment object, the value should be merged and not appended if the name is the same in both.
+			var sameNamedObjects bool
 
 			switch val2 := val2.(type) {
 			case map[string]interface{}:
+				// If the policy value and the current value are different types, use the same logic
+				// as the default case.
+				val1, ok := val1.(map[string]interface{})
+				if !ok {
+					mergedObj = val1
+
+					break
+				}
+
+				if name2, ok := val2["name"].(string); ok && name2 != "" {
+					if name1, ok := val1["name"].(string); ok && name1 == name2 {
+						sameNamedObjects = true
```

Comment:
I was thinking about this over the weekend. I'm wondering if we should go ahead and make this a loop over a package variable array of strings rather than a single hardcoded value. Then it'll be ready to go if/when we support customizing matching keys.

Reply:
It's an interesting idea, but I think that falls under the "premature optimization" category.  :smile: